### ISSUE TITLE
Project architecture

### ISSUE THREAD

**Issue by ihippik** [greazleay/vehicle-api#13]
Your project looks like a good PHP application :)

But I don't like this architecture at all.
Those. we don't usually group files by their functionality.
For large projects, the DDD approach is best.

That is, for example, you have the entity User - you make a package of the same name and put both controllers and DTO models in it etc.

**Comment by greazleay** [comment]
Yeah X-D, some of my projects actually start out this way during the initial learning phase. But I eventually make lots of modifications over time e.g. this project https://github.com/greazleay/members-only-api which uses an architecture very close to DDD. Even this project has been modified and improved upon a lot since I started it.

I would make adjustments to the architecture too.

**Comment by greazleay** [comment]
The architecture has been changed to follow Domain Driven Design (DDD) pattern, though I saw many conflicting implementations of this pattern I had to settle for the one I felt most comfortable with. 

This would be like a first draft of the DDD architecture change, I would keep reading about this pattern and make further adjustments in the near future. Also, the src folder has been removed
